### ISSUE TITLE
Fizzle Snapshot Card order

### ISSUE THREAD

**Issue by hadrien-g** [batstyx/Graveyard#35]
Hello !
First, thanks for this plugin, this is what I was looking for fizzle snapshot on HDT, since it's natively missing :)
But I noticed cards in fizzle snapshot are in a random order, would it be possible for you to modify it so they are showing in the correct order (from the left to the right in the hand at the moment where it's snapshoted)

Best regards, Thanks

**Comment by batstyx** [comment]
I played exactly one game to test this change, so good luck to you :)

**Comment by hadrien-g** [comment]
Hello, thanks !

tested it, it nearly worked as intented, cards are in correct order except when you have 2 copy of a card in the hand, it will group them together.

example with 2 conductivity in my hand

here the correct order was :
Novice zapper / kobold geomancer / Bloodmage / **conductivity** / Flash of lightning  / **conductivity** / crash of thunder / pop-up book
and it shows on the plugin
Novice zapper / kobold geomancer / Bloodmage / **conductivity / conductivity** / Flash of lightning / crash of thunder / pop-up book

![fizzle](https://github.com/batstyx/Graveyard/assets/154736034/535e8f58-04cb-41ae-b5db-92cc3845f4d8)

**Comment by batstyx** [comment]
I really hoped that simple addition would do it. I could have sworn I had duplicate cards correctly ordered in the test I did but the screenshot didn't save for some reason so I can't check.
Apart from Demon Hunter Outcast cards, I'm assuming this is mainly a cosmetic annoyance?

**Comment by hadrien-g** [comment]
Hello ! 
Sadly, no, it's not cosmetic, order is really important, because you don't often have all the room in your hand to get every card from the snapshot. 
If I have 8 cards on my snapshot but only 5 spot on my hand and need a precise card which show on 5th spot in the list but is in reality on the 6th spot, it can instantly loose the game.

**Comment by batstyx** [comment]
Oops. I do, ofc, "know" this, but I'd forgotten. I don't think I've ever played a tight enough strategy that I've lost this way!
If I get the time I'll take a further look at this. It may be I'm using the HDT API wrong. However if it's an HDT bug then, unless it's straightforward, it's well above my pay grade.

**Comment by hadrien-g** [comment]
True, it's for some specific deck/strategy,I admit, it's not for every game :)
Thanks, it's allready awesome that you improve the order !

**Comment by batstyx** [comment]
Related:
HearthSim/Hearthstone-Deck-Tracker#4547
HearthSim/Hearthstone-Deck-Tracker#4503

**Comment by batstyx** [comment]
https://github.com/HearthSim/Hearthstone-Deck-Tracker/issues/4503#issuecomment-2736034037
